ansible/deploy_site_playbook.yml

```diff
@@ -3,38 +3,74 @@
 -
   hosts: all
   tasks:
-    # Copy docker-compose and any supporting files
+    ####
+    # Setup the docker containers & docker-compose based environment
+    ####
     - name: Ensures ~/docker-files/keys/ dirs exist
       file:
         path: "{{ item }}"
         state: directory
       loop:
         - ~/docker-files
         - ~/docker-files/keys
+      notify: Restart docker containers
     - name: Copy ~/docker-files/ Files
       copy:
         src: ./docker-files/{{ item }}
         dest: ~/docker-files/{{ item }}
       loop:
         - docker-compose.yml
+      notify: Restart docker containers
     - name: Copy PEM key
       copy:
         src: ./docker-files/keys/{{ env }}-1nfinite.pem
         dest: ~/docker-files/keys/1nfinite.pem
+      notify: Restart docker containers
     - name: Create secrets file for containers
       template:
         src: docker-files/{{ item }}.j2
         dest: ~/docker-files/{{ item }}
       loop:
         - web-portal.env
         - api.env
+      notify: Restart docker containers
```

Comment:
Will this cause issues on first deploy with the cert-bot stuff. I forget all the details, I think what happens is if you configure nginx the ways it's supposed to be configured it bombs because cert bot has not created the pem files yet.

At the same time you can't create them because certbot uses nginx to validate when creating the certs so you get stuck in a loop.


Maybe this isn't an issue. This is the second playbook we run so maybe at this time we've already ran certbot, just thinking outloud and double checking :-)

Reply:
Restarting the docker containers won't be affected by certbot, but I'll take a closer look at the nginx aspects to understand the impact.  I suspect that addressing it will be more a matter for documentation of initial / manual setup procedures than changing this playbook.